beam/src/components/BeamFilterOption.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+	<BeamHeading class="beam_filter-option-heading">{{ title }}</BeamHeading>
+	<div @click="toggle" class="beam_filter-option">
+		<div ref="select" class="beam_filter-option-select">
+			<div class="beam_filter-arrow">
+				<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 35.36 70.71">
+					<polygon points="0 70.71 0 0 35.36 35.36 0 70.71" />
+				</svg>
+			</div>
+			<div class="beam_filter-label">
+				<label>{{ choice }}</label>
+			</div>
+		</div>
+		<ul ref="menu" v-if="open" class="beam_filter-select-menu">
+			<li
+				v-for="(opt, index) in choices"
+				@click="updateValue(opt)"
+				class="beam_filter-select-option"
+				:class="{ selected: choice == opt.choice }"
+				:data-value="opt.value"
+				:key="index">
+				{{ opt.choice }}
+			</li>
+		</ul>
+	</div>
+</template>
+<script setup>
+import { ref, defineProps } from 'vue'
+
+const props = defineProps({
+	title: {
+		type: String,
+		default: 'title',
+	},
+	choices: {
+		type: [Array, { choice: String, value: String }],
+	},
+})
```

Comment:
This is mostly syntactical sugar, but this could be rewritten as:

```suggestion
const { title = 'title', choices = [] } = defineProps<{
	title?: string
	choices: { choice: string; value: string }[]
}>()
```

This would also require the `<script>` tag to change to `<script setup lang="ts">`